cpp/include/cudf/ast/detail/linearizer.hpp

```diff
@@ -103,10 +103,24 @@ class linearizer {
   /**
    * @brief Construct a new linearizer object
    *
+   * @param expr The expression to create an evaluable linearizer for.
+   * @param left The left table used for evaluating the abstract syntax tree.
+   * @param right The right table used for evaluating the abstract syntax tree.
+   */
+  linearizer(detail::node const& expr, cudf::table_view left, cudf::table_view right)
+    : _left(left), _right(right), _node_count(0), _intermediate_counter()
```

Comment:
Braces in the initialization list are preferrable because they catch implicit conversions
```suggestion
    : _left{left}, _right{right}, _node_count{0}, _intermediate_counter{}
```